core/modules/taxonomy/plugins/access/taxonomy_term_depth_access.inc

```diff
@@ -0,0 +1,97 @@
+<?php
+  /**
+   * @file
+   * Plugin to provide access control based upon taxonomy term depth.
+   */
+  class TaxonomyTermDepthLayoutAccess extends LayoutAccess {
+    /**
+     * Constructor for a Layout access rule.
+     */
+    function __construct($plugin_name, array $data = array()) {
+      parent::__construct($plugin_name, $data);
+      $this->settings += array(
+        'depth' => array()
+      );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    function form(&$form, &$form_state) {
+      parent::form($form, $form_state);
+
+      $form['depth'] = array(
+        '#type' => 'number',
+        '#title' => t('Depth'),
+        '#description' => t('Depth is the level of nesting for a taxonomy term. See example below.'),
+        '#default_value' => $this->settings['depth'],
+      );
+
+      $code  = '<code>';
+      $code .= 'State (1st level, <strong>depth 1</strong>)<br />';
+      $code .= '- City (2nd level, <strong>depth 2</strong>)<br />';
+      $code .= '&nbsp; - Neighborhood (3rd level, <strong>depth 3</strong>)';
+      $code .= '</code>';
+
+      $form['depth_example'] = array(
+        '#type' => 'markup',
+        '#markup' => $code,
+      );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    function formSubmit($form, &$form_state) {
+      parent::formSubmit($form, $form_state);
+      $this->settings['depth'] = $form_state['values']['depth'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    function summary() {
+      if (isset($this->settings['depth'])) {
+        return t('Taxonomy term depth: @depth', array('@depth' => $this->settings['depth']));
+      }
+
+      return '';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    function checkAccess() {
+      if (!isset($this->contexts['taxonomy_term']->data)) {
+        return FALSE;
+      };
+
+      $tid = $this->contexts['taxonomy_term']->data->tid;
+      $depth = $this->GetTermDepth($tid);
+
+      if ($depth == $this->settings['depth']) {
+        return TRUE;
+      }
+
+      return FALSE;
+    }
+
+    /**
+     * Gets the depth of a term.
+     *
+     * @param $tid
+     *   The taxonomy term ID
+     *
+     * @return int
+     *   An integer representing how many levels deep your term is nested.
+     */
+    function GetTermDepth($tid) {
```

Comment:
functions are supposed to be lower camel case, so `getTermDepth`.

And also: none of the methods declares visibility - example: `public function getTermDepth(` or `protected function getTermDepth(`...

Reply:
Fixed camelcase, but I don't know anything about methods declaring visibility. I will either need more help with this OR someone else might need to either take over this PR or submit a PR against this PR. Either works for me. 

Maybe I'll ask for help on this during office hours. ;-)

Reply:
An easier approach: use `public` for all. The parent class (LayoutAccess) doesn't declare visibility, too. If not set, it defaults to "public".

So you could also skip declaring, but it's usually best practice to do - even if it's "public" (the default).

I'm not sure if that documentation might help you: https://www.php.net/manual/en/language.oop5.visibility.php#language.oop5.visiblity-methods

Reply:
Well, I'm an complete newbie to working in OOP. But, I watched enough videos today and read enough to understand the purpose of declaring visibility. However, I didn't write this code and am still not entirely sure if any of these functions should be private or protected. 

I went with declaring them all public: 1) because @indigoxela hinted at that 2) because by default that is what they would have been anyway. 

I'd love feedback on this and/or information that would help me figure what the correct visibility should be.

Reply:
I agree about setting the visibility. Let's set `getTermDepth` to protected right off the bat since it should only be accessed by this class or child classes. It's harder to restrict in the future compared to opening up.